Automattic/jetpack
# Assets: Add textdomain aliasing

#### Changes proposed in this Pull Request:

This adds the ability to register textdomain aliases with the Assets
package.

Assets then hooks into `__()` and so on to have them check the
aliased-to domain if no translation is found for the aliased-from
domain. It also hooks into script translation file loading to load a
translation file for the aliased-to domain if no file exists for the
aliased-from domain.

This brings us one step closer to fixing #21690. Followups will switch
all the packages to use alias-able textdomains (instead of them all
using "jetpack") and arrange for the aliases to actually be registered.

#### Jetpack product discussion


#21690

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is CI happy?
* Testing this will probably be a bit tricky. This may work:
  * Set your site to a non-English locale.
  * Verify that the Jetpack dashboard is translated.
  * Move the files in `wp-content/languages/plugins/jetpack*` to `wp-content/languages/plugins/xxxjetpack*`
  * Verify that the Jetpack dashboard is no longer showing translations.
  * Edit `wp-content/plugins/jetpack/vendor/autoload_packages.php` to add this at the end to alias "jetpack" to "xxxjetpack":
    ```php
    \Automattic\Jetpack\Assets::alias_textdomain( 'jetpack', 'xxxjetpack', 'plugins', '1.2.3' );
    ```
  * Verify that the Jetpack dashboard is translated again.
